docs/section-7/odometry-kinematics.md

```diff
@@ -0,0 +1,57 @@
+## What is Odometry?
+Odometry is the use of sensor information to estimate the robot's position on the field. Odometry works for both differential drivetrains (think arcade/tank drive) and swerve drivetrains (covered in a later section - drivetrains that can move forward, strafe, and rotate, perhaps at the same time).
+
+## WPILib Odometry Class
+Before we cover odometry, we first need to cover the `Translation2d`, `Rotation2d`, and `Pose2d` objects.
+
+`Translation2d` stores x and y positions of the robot in meters as well as has several other useful methods, including scaling using `plus()`, `rotateBy()`, `times()`, and `getNorm()` (normalization a.k.a the unit vector on the unit circle in the same direction as the point).
+
+`Rotation2d` stores an angular position value in radians as well as has several other useful methods, including `getDegrees()`, `fromDegrees()`, `rotateBy()`, and `plus()`.
+
+`Pose2d` is a combination of `Translation2d` and `Rotation2d` into a singular class, alongside methods such as `plus()` and `relativeTo()`.
+
+To declare an odometry object, you need to specify the robot's pose and heading. For DifferentialDriveOdometry, you can do this as follows:
+
+```
+Rotation2d heading = /* Get gyro heading */;
+// Create odometry at (0, 0)
+DifferentialDriveOdometry odometry1 = new DifferentialDriveOdometry(heading)
+// Or...
+// Create odometry at (x, y)
+DifferentialDriveOdometry odometry2 = new DifferentialDriveOdometry(heading, new Pose2d(x, y, heading))
+```
+
+The odometry object has three methods of importance:
+
+- `getPoseMeters()` = returns a Pose2d object representing the robot's current position on the field.
```

Comment:
position *and rotation*.

Btw, is the rotation angle in the returned `Pose2D` relative to the heading that was passed to the `DifferentialDriveOdometry` constructor or is it absolute. For example, does this print 0 or 90?

```
DifferentialDriveOdometry odometry = new DifferentialDriveOdometry(Rotation2D.fromDegrees(90));
System.out.println(odometry.getPoseMeters().getRotation().toDegrees());
```

Whatever the answer, your explanation should make it clear.

Reply:
I believe that it will print zero because no pose was specified. Looking at the constructor of DifferentialDriveOdometry, the initial heading is used to find the offset between the initial pose (in this case (0, 0) facing towards 0 degrees) and the current heading. The gyro heading does not appear to update the Pose2D object in the constructor.

Reply:
I agree. Just make sure this is clear in the docs. This is a great example of something that is currently ambiguous in the official docs that you can clarify in our training material.